src/LinuxTracing/TracerThread.cpp

```diff
@@ -507,17 +507,17 @@ bool TracerThread::OpenInstrumentedTracepoints(const std::vector<int32_t>& cpus)
   return !tracepoint_event_open_errors;
 }
 
-void TracerThread::Run(const std::shared_ptr<std::atomic<bool>>& exit_requested) {
-  FAIL_IF(listener_ == nullptr, "No listener set");
-
+void TracerThread::Startup() {
+  ORBIT_SCOPE_FUNCTION;
   Reset();
 
   // perf_event_open refers to cores as "CPUs".
 
   // Record context switches from all cores for all processes.
-  std::vector<int32_t> all_cpus;
-  for (int32_t cpu = 0; cpu < GetNumCores(); ++cpu) {
-    all_cpus.push_back(cpu);
+  int32_t number_of_cores = GetNumCores();
+  std::vector<int32_t> all_cpus(number_of_cores);
+  for (int32_t cpu = 0; cpu < number_of_cores; ++cpu) {
+    all_cpus[cpu] = cpu;
```

Comment:
I don't really like this, I prefer `reserve` and `push_back` to 0-initialization and reassigning.

Reply:
done.